Bug: GitHub config can't be copied

```python
12-09 23:47:09 [ERROR] nonebot | Running Matcher(type='', module=src.plugins.github.plugins.config, lineno=143) failed.
Traceback (most recent call last):
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\runpy.py", line 198, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\runpy.py", line 88, in _run_code
    exec(code, run_globals)
  File "c:\Users\hmy01\.vscode\extensions\ms-python.debugpy-2024.13.2024112901-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher/../..\debugpy\__main__.py", line 71, in <module>
    cli.main()
  File "c:\Users\hmy01\.vscode\extensions\ms-python.debugpy-2024.13.2024112901-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher/../..\debugpy/..\debugpy\server\cli.py", line 501, in main
    run()
  File "c:\Users\hmy01\.vscode\extensions\ms-python.debugpy-2024.13.2024112901-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher/../..\debugpy/..\debugpy\server\cli.py", line 351, in run_file
    runpy.run_path(target, run_name="__main__")
  File "c:\Users\hmy01\.vscode\extensions\ms-python.debugpy-2024.13.2024112901-win32-x64\bundled\libs\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 310, in run_path
    return _run_module_code(code, init_globals, run_name, pkg_name=pkg_name, script_name=fname)
  File "c:\Users\hmy01\.vscode\extensions\ms-python.debugpy-2024.13.2024112901-win32-x64\bundled\libs\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 127, in _run_module_code
    _run_code(code, mod_globals, init_globals, mod_name, mod_spec, pkg_name, script_name)
  File "c:\Users\hmy01\.vscode\extensions\ms-python.debugpy-2024.13.2024112901-win32-x64\bundled\libs\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 118, in _run_code
    exec(code, run_globals)
  File "c:\Users\hmy01\.vscode\extensions\ms-python.python-2024.20.0-win32-x64\python_files\vscode_pytest\run_pytest_script.py", line 64, in <module>
    pytest.main(arg_array)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\config\__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\main.py", line 330, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\main.py", line 283, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\main.py", line 337, in _main
    config.hook.pytest_runtestloop(session=session)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\main.py", line 362, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\runner.py", line 113, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\runner.py", line 132, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\runner.py", line 241, in call_and_report
    call = CallInfo.from_call(
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\runner.py", line 341, in from_call
    result: TResult | None = func()
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\runner.py", line 242, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\runner.py", line 174, in pytest_runtest_call
    item.runtest()
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pytest_asyncio\plugin.py", line 457, in runtest
    super().runtest()
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pluggy\_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\_pytest\python.py", line 159, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\pytest_asyncio\plugin.py", line 929, in inner
    _loop.run_until_complete(task)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\asyncio\base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\asyncio\windows_events.py", line 322, in run_forever
    super().run_forever()
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\asyncio\base_events.py", line 641, in run_forever
    self._run_once()
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\asyncio\base_events.py", line 1986, in _run_once
    handle._run()
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\utils.py", line 254, in run_coro_with_shield
    return await coro
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\message.py", line 506, in check_and_run_matcher
    await _run_matcher(
> File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\message.py", line 458, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebug\mixin\process\fake.py", line 83, in fake_run
    await run(self, *args, **kwargs)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 900, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebug\mixin\process\fake.py", line 62, in fake_simple_run
    await simple_run(self, *args, **kwargs)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 837, in simple_run
    await handler(
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\dependencies\__init__.py", line 113, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\src\plugins\github\plugins\config\__init__.py", line 153, in handle_auto_merge
    async with bot.as_installation(installation_id):
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\nonebot\adapters\github\bot.py", line 255, in as_installation
    g := self._github.with_auth(
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\githubkit\github.py", line 126, in with_auth
    return GitHub(auth=auth, config=self.config.copy())
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\githubkit\config.py", line 30, in copy
    return self.__class__(**self.dict())
  File "c:\Users\hmy01\Works\Working\Bot\noneflow\.venv\Lib\site-packages\githubkit\config.py", line 27, in dict
    return asdict(self)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\dataclasses.py", line 1329, in asdict
    return _asdict_inner(obj, dict_factory)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\dataclasses.py", line 1339, in _asdict_inner
    f.name: _asdict_inner(getattr(obj, f.name), dict)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\dataclasses.py", line 1386, in _asdict_inner
    return copy.deepcopy(obj)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 162, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 259, in _reconstruct
    state = deepcopy(state, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 136, in deepcopy
    y = copier(x, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 221, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 162, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 259, in _reconstruct
    state = deepcopy(state, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 136, in deepcopy
    y = copier(x, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 221, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 162, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 259, in _reconstruct
    state = deepcopy(state, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 136, in deepcopy
    y = copier(x, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 221, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "C:\Users\hmy01\AppData\Roaming\uv\python\cpython-3.12.7-windows-x86_64-none\Lib\copy.py", line 151, in deepcopy
    rv = reductor(4)
TypeError: cannot pickle '_thread.lock' object
```

--- a/githubkit/config.py
+++ b/githubkit/config.py
@@ -1,4 +1,4 @@
-from dataclasses import asdict, dataclass
+from dataclasses import dataclass, fields
 from typing import Any, Optional, Union
 from typing_extensions import Self
 
@@ -24,9 +24,11 @@ class Config:
     rest_api_validate_body: bool
 
     def dict(self) -> dict[str, Any]:
-        return asdict(self)
+        """Return the config as a dictionary without copy values."""
+        return {field.name: getattr(self, field.name) for field in fields(self)}
 
     def copy(self) -> Self:
+        """Return a shallow copy of the config."""
         return self.__class__(**self.dict())
 
 
@@ -107,6 +109,7 @@ def get_config(
     auto_retry: Union[bool, RetryDecisionFunc] = True,
     rest_api_validate_body: bool = True,
 ) -> Config:
+    """Build the configs from the given options."""
     return Config(
         build_base_url(base_url),
         build_accept(accept_format, previews),

--- a/tests/test_auth/test_auth_switch.py
+++ b/tests/test_auth/test_auth_switch.py
@@ -1,0 +1,9 @@
+from githubkit import GitHub, UnauthAuthStrategy
+
+
+def test_auth_switch(g: GitHub):
+    new_auth = UnauthAuthStrategy()
+
+    new_g = g.with_auth(new_auth)
+    assert new_g.auth is new_auth
+    assert new_g.config == g.config

